core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,288 @@
+    await this.clickOn(addNewResponseButton);
+    await this.page.waitForSelector(resonseModalHeaderSelector, {hidden: true});
+  }
+
+  /**
+   * Function to navigate to the preview tab.
+   */
+  async navigateToPreviewTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileNavbarDropdown);
+      await this.page.waitForSelector(mobileNavbarPane);
+      await this.clickOn(mobilePreviewTabButton);
+      await this.page.waitForNavigation();
+    } else {
+      await this.clickOn(previewTabButton);
+      await this.page.waitForNavigation();
+    }
+  }
+
+  /**
+   * Function to verify if the exploration is loading correctly in
+   * the preview tab or not via checking the content of the Introduction(first) card.
+   * @param {string} text - The expected introduction card text.
+   */
+  async expectCardContentToBe(text: string): Promise<void> {
+    await this.page.waitForSelector(stateConversationContent, {
+      visible: true,
+    });
+    const element = await this.page.$(stateConversationContent);
+    const introMessage = await this.page.evaluate(
+      element => element.textContent,
+      element
+    );
+    if (introMessage === text) {
+      showMessage(
+        'Preview is on the Introduction card and is loading correctly.'
+      );
+    } else {
+      throw new Error(
+        'Preview is not on the Introduction card or is not loading correctly.'
+      );
+    }
+  }
+
+  /**
+   * Functions to complete the exploration in the preview tab.
+   */
+  async continueToNextCard(): Promise<void> {
+    await this.clickOn(nextCardButton);
+  }
+
+  async enterAnswer(answer: string): Promise<void> {
+    await this.type(floatFormInput, answer);
+  }
+
+  async submitAnswer(): Promise<void> {
+    await this.clickOn(submitAnswerButton);
+  }
+
+  /**
+   * Function to verify if the exploration is completed in the preview tab via checking the toast message.
+   * @param {string} message - The expected toast message.
+   */
+  async expectPreviewCompletionToastMessage(message: string): Promise<void> {
+    await this.page.waitForSelector(explorationCompletionToastMessage, {
+      visible: true,
+    });
+    const element = await this.page.$(explorationCompletionToastMessage);
+    const toastMessage = await this.page.evaluate(
+      element => element.textContent,
+      element
+    );
+    if (toastMessage && toastMessage.includes(message)) {
+      showMessage('Exploration has completed successfully');
+    } else {
+      throw new Error('Exploration did not complete successfully');
+    }
+    await this.page.waitForSelector(explorationCompletionToastMessage, {
+      hidden: true,
+    });
+  }
+
+  /**
+   * Function to restart the preview after it has been completed.
+   */
+  async restartPreview(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (element) {
```

Comment:
Can you describe, in a comment, why this element might not exist?

Reply:
when the mobile navigation bar is expanded, it actually, overlaps the restart button making it unclickable. So, checking for its presence and collapsing it. Mentioned the same in a comment.

Can it make a issue?

Reply:
Yes, please file an issue, thanks. (In general, please file an issue any time you see something is not working as expected -- no need to ask beforehand.)

Reply:
Done.

Reply:
@Akhilesh-max Please link to the issue you filed. (It's a good idea to do this in general when this sort of thing comes up in review comments.)